### PR TITLE
Fix messages

### LINKED ISSUE #98: Incorrect functionality of editgradedtest
editgradedtest unable to work dynamically. Requires the 5 core parameters to work (ra1, ra2, midterms, finals, pe).

Ideally, it should be implemented such that we can dynamically editgradedtest.

### PR DESCRIPTION
**Pull request by raydenlim**
Fix messages issue for `Person`

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Person.java [hunk 1]
```diff
@@ -184,8 +184,7 @@ public String toString() {
                 .add("phone", phone)
                 .add("email", email)
                 .add("telegramHandle", telegramHandle)
-                .add("tags", tags)
-                .add("gradedTests", gradedTests)
+                .add("tags", tags).add("gradedTests", gradedTests)
```
Comment (wesho1107) [inline src/main/java/seedu/address/model/person/Person.java]: Not consistent across the function

**Review by wesho1107 — COMMENTED** [review]
Other than a few nits, LGTM

File: src/test/java/seedu/address/testutil/PersonUtil.java [hunk 2]
```diff
@@ -62,6 +63,14 @@ public static String getEditPersonDescriptorDetails(EditPersonDescriptor descrip
                 tags.forEach(s -> sb.append(PREFIX_TAG).append(s.tagName).append(" "));
             }
         }
+        if (descriptor.getGradedTests().isPresent()) {
+            Set<GradedTest> gradedTests = descriptor.getGradedTests().get();
```
Comment (wesho1107) [inline src/test/java/seedu/address/testutil/PersonUtil.java:67]: Is this preferred over having the descriptor to return the Set<GradedTest> instead for SLAP?

**Review by wesho1107 — COMMENTED** [review]
Other than a few nits, LGTM!

File: src/main/java/seedu/address/model/person/Person.java (hunk 1 above)
Comment (raydenlim) [reply src/main/java/seedu/address/model/person/Person.java]: Woops! Must have been a slip of hand :P

**Review by wesho1107 — APPROVED** [review]
LGTM!

File: src/main/java/seedu/address/logic/parser/EditCommandParser.java [hunk 3]
```diff
@@ -125,4 +130,20 @@ private Optional<Set<Tag>> parseTagsForEdit(Collection<String> tags) throws Pars
...
+     */
+    private Optional<Set<GradedTest>> parseGradedTestForEdit(Collection<String> gradedTest) throws ParseException {
+        assert gradedTest != null;
+
+        if (gradedTest.isEmpty()) {
+            return Optional.empty();
+        }
+        Collection<String> gradedTestSet = gradedTest.size() == 1
+                && gradedTest.contains("") ? Collections.emptySet() : gradedTest;
+        return Optional.of(ParserUtil.parseGradedTests(gradedTestSet));
```
Comment (wesho1107) [inline src/main/java/seedu/address/logic/parser/EditCommandParser.java]: Missing test case to cover for this.

**Review by wesho1107 — CHANGES_REQUESTED** [review]
Left out on one test case for the Parser! Other than that, LGTM

**Comment by raydenlim** [comment]
Updated the following classes:
`AddCommand`
`EditCommand`
`EditCommandParser`
`ParserUtil`
`GradedTest`
`typicalPersonAddressBook`
`typicalConsultationList`
`typicalSessionList`
`CommandTestUtil`
`AddressBookParserTest`
`EditCommandParserTest`
`GradedTestTest`
`PersonTest`
`PersonUtil`
`TypicalPersons`

**Comment by raydenlim** [comment]
Fix Bugs and Logic
Updated Json files

**Comment by raydenlim** [comment]
Update logic for parsing `gradedtest`
Add more testcases for `parserutiltest`

**Review by wesho1107 — APPROVED** [review]
LGTM!

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 4]
```diff
@@ -279,32 +280,98 @@ public static TaskDescription parseTaskDescription(String name) throws ParseExce
...
+                MidTerms midTerms = new MidTerms(gradedTestParser(trimmedGradedTest)[2]);
+                Finals finals = new Finals(gradedTestParser(trimmedGradedTest)[3]);
+                PracticalExam practicalExam = new PracticalExam(gradedTestParser(trimmedGradedTest)[4]);
+
+                return new GradedTest(readingAssessment1, readingAssessment2, midTerms, finals, practicalExam);
+            } catch (ParseException e) {
+                if (!GradedTest.isValidGradeTestName(trimmedGradedTest)) {
+                    throw new ParseException(GradedTest.MESSAGE_CONSTRAINTS);
+                }
+                return new GradedTest(trimmedGradedTest);
```
Comment (ldinghan) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:304]: test case

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 5]
```diff
@@ -279,32 +280,98 @@ public static TaskDescription parseTaskDescription(String name) throws ParseExce
...
+        String[] scores = new String[5];
 
-            return new GradedTest(readingAssessment1, readingAssessment2, midTerms, finals, practicalExam);
-        } catch (ParseException e) {
-            if (!GradedTest.isValidGradeTestName(trimmedGradedTest)) {
-                throw new ParseException(GradedTest.MESSAGE_CONSTRAINTS);
+        for (int i = 0; i < components.length; i++) {
+            String[] fieldParts = components[i].split(":");
+            if (fieldParts.length != 2) {
+                throw new ParseException("Invalid GradedTest format for field " + i);
```
Comment (ldinghan) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:328]: test case

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 6]
```diff
@@ -279,32 +280,98 @@ public static TaskDescription parseTaskDescription(String name) throws ParseExce
...
+            }
+            break;
+        case "PE":
+            if (!fieldValue.matches(PracticalExam.VALIDATION_REGEX)) {
+                throw new ParseException(PracticalExam.MESSAGE_CONSTRAINTS);
             }
-            return new GradedTest(trimmedGradedTest);
+            break;
+        default:
+            throw new ParseException("Invalid field name: " + fieldName);
```
Comment (ldinghan) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:374]: test case

File: src/main/java/seedu/address/model/gradedtest/GradedTest.java [hunk 7]
```diff
@@ -78,12 +90,56 @@ private String[] parseGradedTest(String gradedTestsIndv) throws ParseException {
...
+        for (int i = 0; i < components.length; i++) {
+            String[] fieldParts = components[i].split(":");
+            if (fieldParts.length != 2) {
+                throw new ParseException("Invalid GradedTest format for field " + i);
+            }
+            String fieldName = fieldParts[0].trim();
+            String fieldValue = fieldParts[1].trim();
+            String errorMessage = validateField(fieldName, fieldValue);
+            if (errorMessage != null) {
+                throw new ParseException(errorMessage);
```
Comment (ldinghan) [inline src/main/java/seedu/address/model/gradedtest/GradedTest.java:105]: update gradedtesttest for all lines not covered by tests

**Review by ldinghan — APPROVED** [review]
LGTM, just need to add in test cases for ParserUtil and GradedTest

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 4 above)
Comment (raydenlim) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:304]: Codecov is confusing :(
doesnt this test already cover that line?

```
    @Test
    public void parseGradedTest_invalidFormat_throwsParseException() {
        String input = "RA1:10 | RA2:20 | MidTerms:30 | Finals:40 | PE:50 | ExtraComponent:60";
        assertThrows(ParseException.class, () -> ParserUtil.parseGradedTest(input));
    }
```

**Pull request merged**